src/main/java/Byte/Byte.java

```diff
@@ -0,0 +1,266 @@
+            System.out.println("Sorry, I can only handle up to " + MAX_TASKS + " tasks!");
+        }
+        printLineSeparator();
+    }
+
+    private static void listTasks() {
+        for (int i = 0; i < tasksList.size(); i++) {
+            System.out.println((i + 1) + ". " + tasksList.get(i));
+        }
+        printLineSeparator();
+    }
+
+    private static void markTask(String taskNumberStr, boolean isDone) {
+        int taskNumber;
+        try {
+            taskNumber = Integer.parseInt(taskNumberStr)-1;
+            Task task = tasksList.get(taskNumber);
+            if (isDone) {
+                task.markAsDone();
+                System.out.println("Nice! I've marked this task as done:\n  " + task);
+            } else {
+                task.markAsNotDone();
+                System.out.println("OK, I've marked this task as not done yet:\n  " + task);
+            }
+            tasksChanged = true;
+        } catch (NumberFormatException e) {
+            System.out.println("Please enter a valid task number.");
+        } catch (IndexOutOfBoundsException e) {
+            System.out.println(e.getMessage());
+        }
+        printLineSeparator();
+    }
+
+    private static void printLineSeparator() {
+        System.out.println("____________________________________________________________");
+    }
+
+    private static void deleteTask(String userInput) throws ByteException {
+        try {
+            int taskNumber = Integer.parseInt(userInput.substring("delete ".length())) - 1;
+            if (taskNumber < 0 || taskNumber >= tasksList.size()) {
+                throw new IndexOutOfBoundsException("Task number is out of range");
+            }
+            Task deletedTask = tasksList.remove(taskNumber);
+            System.out.println("Noted. I've removed this task:\n  " + deletedTask);
+            System.out.println("Now you have " + tasksList.size() + " tasks in the list.");
+        } catch (NumberFormatException e) {
+            throw new ByteException("Please enter a valid task number.");
+        } catch (IndexOutOfBoundsException e) {
+            throw new ByteException(e.getMessage());
+        }
+        printLineSeparator();
+    }
+
+    private static void loadTasksFromFile(String filePath) throws FileNotFoundException {
+        File file = new File(filePath);
+        createParentDirectory(file);
+
+        Scanner scanner = new Scanner(file);
+        while (scanner.hasNext()) {
+            String line = scanner.nextLine();
+            processTaskFromLine(line);
+        }
+    }
+
+    private static void processTaskFromLine(String line) {
+        String[] parts = line.split(" \\| ");
+        String type = parts[TASK_TYPE_INDEX];
+        String description = parts[DESCRIPTION_INDEX];
+        switch (type) {
+            case "T":
+                addTask(new ToDo(description));
+                break;
+            case "D":
+                String by = parts[DEADLINE_TIME_INDEX ];
```

Comment:
```suggestion
                String by = parts[DEADLINE_TIME_INDEX];
```
Perhaps delete the space here to be consistent with other part of the code.

Reply:
Noted